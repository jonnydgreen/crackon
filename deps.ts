// Deno.land
export { EOL } from "https://deno.land/std@0.114.0/fs/eol.ts";
export {
  dirname,
  fromFileUrl,
  join,
} from "https://deno.land/std@0.114.0/path/mod.ts";
import "https://deno.land/x/reflection@0.0.2/mod.ts";
export { Command } from "https://deno.land/x/cliffy@v0.20.1/command/mod.ts";

// Skypack CDN
export {
  extendSchema,
  GraphQLEnumType,
  GraphQLError,
  GraphQLInputObjectType,
  GraphQLInterfaceType,
  GraphQLList,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLScalarType,
  GraphQLSchema,
  GraphQLUnionType,
  isOutputType,
  isRequiredArgument,
  isTypeExtensionNode,
  Kind,
  parse,
} from "https://cdn.skypack.dev/graphql@v16.0.1?dts";
export type {
  DefinitionNode,
  GraphQLArgument,
  GraphQLField,
  GraphQLInputField,
  GraphQLInputType,
  GraphQLNamedType,
  GraphQLOutputType,
} from "https://cdn.skypack.dev/graphql@v16.0.1?dts";
export {
  Container,
  inject,
  injectable,
} from "https://cdn.skypack.dev/inversify@v6.0.1?dts";

// JSPM
// @deno-types=https://dev.jspm.io/typescript/lib/typescript.d.ts
export { default as ts } from "https://dev.jspm.io/typescript";

// Dev Deps
export {
  assert,
  assertEquals,
  assertRejects,
  assertStrictEquals,
  assertThrows,
} from "https://deno.land/std@0.114.0/testing/asserts.ts";
