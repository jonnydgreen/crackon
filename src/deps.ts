// Deno.land
export { Command } from "https://deno.land/x/cliffy@v0.20.1/command/mod.ts";
export { EOL } from "https://deno.land/std@0.114.0/fs/eol.ts";
export {
  dirname,
  fromFileUrl,
  join,
} from "https://deno.land/std@0.114.0/path/mod.ts";

// Skypack CDN
export {
  extendSchema,
  GraphQLEnumType,
  GraphQLError,
  GraphQLInputObjectType,
  GraphQLInterfaceType,
  GraphQLList,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLScalarType,
  GraphQLSchema,
  GraphQLUnionType,
  isOutputType,
  isRequiredArgument,
  isTypeExtensionNode,
  Kind,
  parse,
} from "https://cdn.skypack.dev/graphql@v16.0.1?dts";
export type {
  DefinitionNode,
  GraphQLArgument,
  GraphQLField,
  GraphQLInputField,
  GraphQLInputType,
  GraphQLNamedType,
  GraphQLOutputType,
} from "https://cdn.skypack.dev/graphql@v16.0.1?dts";
export { default as Handlebars } from "https://cdn.skypack.dev/handlebars@4.7.7?dts";
