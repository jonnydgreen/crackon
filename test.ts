import zlib from "https://deno.land/std@0.145.0/node/zlib.ts";

console.log(zlib.codes);
