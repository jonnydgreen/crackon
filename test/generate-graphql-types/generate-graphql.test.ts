import { assertEquals, assertSnapshot, Mocking, Testing } from "test.deps";
import { Colours } from "deps";
import { cli } from "cli/index.ts";

Testing.describe("Generate GraphQL", () => {
  Testing.beforeEach(() => {
    Mocking.restore();
  });

  [
    {
      name:
        "should display help text to stderr when no arguments or flags are passed",
      args: ["graphql"],
    },
    {
      name: "should display help text to stderr when passing the '--help' flag",
      args: ["graphql", "--help"],
    },
    {
      name: "should display help text to stderr when passing the '-h' flag",
      args: ["graphql", "-h"],
    },
  ].forEach((definition) => {
    Testing.it(definition.name, async (t) => {
      // Arrange
      const args = definition.args;
      const consoleErrorSpy = Mocking.stub(console, "error");
      const consoleLogSpy = Mocking.stub(console, "log");

      // Act
      await cli(args);

      // Assert
      assertEquals(
        consoleLogSpy.callCount,
        0,
        `console.log should not be called. Called ${consoleLogSpy.callCount} times`,
      );
      assertEquals(
        consoleErrorSpy.callCount,
        1,
        `console.error should be called once. Called ${consoleErrorSpy.callCount} times`,
      );
      assertEquals(
        consoleErrorSpy.getCall(0).args.length,
        1,
        `console.error should be called with a single argument. Called with ${
          consoleErrorSpy.getCall(0).args.length
        } arguments`,
      );
      await assertSnapshot(
        t,
        Colours.stripColor(consoleErrorSpy.getCall(0).args[0]),
      );
    });
  });
});
