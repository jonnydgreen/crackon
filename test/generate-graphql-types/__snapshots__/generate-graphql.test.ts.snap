export const snapshot = {};

snapshot[`Generate GraphQL > should display help text to stderr when no arguments or flags are passed 1`] = `
'
  Usage:   hello-gen graphql
  Version: 1.0.0            

  Description:

    GraphQL Generation functionality using an input schema

  Options:

    -l, --log-level  <level>   - Set log level.                                         (Default: "info")
    -f, --format     <format>  - Set the format for printing command output resources.  (Default: "json")
    -h, --help                 - Print help information.                                                 

  Commands:

    types  - Generates GraphQL Types from an input schema

  Environment variables:

    HELLO_GEN_LOG_LEVEL  <level>   - Set log level.                                       
    HELLO_GEN_FORMAT     <format>  - Set the format for printing command output resources.
'
`;

snapshot[`Generate GraphQL > should display help text to stderr when passing the '--help' flag 1`] = `
'
  Usage:   hello-gen graphql
  Version: 1.0.0            

  Description:

    GraphQL Generation functionality using an input schema

  Options:

    -l, --log-level  <level>   - Set log level.                                         (Default: "info")
    -f, --format     <format>  - Set the format for printing command output resources.  (Default: "json")
    -h, --help                 - Print help information.                                                 

  Commands:

    types  - Generates GraphQL Types from an input schema

  Environment variables:

    HELLO_GEN_LOG_LEVEL  <level>   - Set log level.                                       
    HELLO_GEN_FORMAT     <format>  - Set the format for printing command output resources.
'
`;

snapshot[`Generate GraphQL > should display help text to stderr when passing the '-h' flag 1`] = `
'
  Usage:   hello-gen graphql
  Version: 1.0.0            

  Description:

    GraphQL Generation functionality using an input schema

  Options:

    -l, --log-level  <level>   - Set log level.                                         (Default: "info")
    -f, --format     <format>  - Set the format for printing command output resources.  (Default: "json")
    -h, --help                 - Print help information.                                                 

  Commands:

    types  - Generates GraphQL Types from an input schema

  Environment variables:

    HELLO_GEN_LOG_LEVEL  <level>   - Set log level.                                       
    HELLO_GEN_FORMAT     <format>  - Set the format for printing command output resources.
'
`;
